### ISSUE TITLE
Trojan:Win32/Tisifi.B Virus Detected

### ISSUE THREAD

**Issue by UsagiIshiko** [schellingb/WiiBalanceScale#1]
When I tried to download your first release of the software to try out on my pc, Windows Defender automatically removed the file due to a virus being detected.

![image](https://user-images.githubusercontent.com/8121917/54081717-7e79d580-434c-11e9-8d67-046fa5edf30e.png)

**Comment by schellingb** [comment]
That's a harsh jdugement there Windows Defender ...

Here's the output of Virus Total:

File|Link
-|-
ZIP|https://www.virustotal.com/#/file/69267f5861949b64bb0f994ba4d8a063f1e995ba9fb842aa4cbcc0a0f3ca9d9a
EXE|https://www.virustotal.com/#/file/748c3f9b38bff314451d87184a3dd8a980d2ec62041cc150bc36387e2d6bb480

Weird that according to Virus Total the Microsoft virus scanner does not complain.
Let's hope it's just a temporary false warning. I could try to recompile it with different settings but sometimes it's hard to please every scanner.

**Comment by UsagiIshiko** [comment]
Fair enough. I will try it again after an update and restart to see if it changes anything.

**Comment by RichyT** [comment]
Same thing here.

**Comment by freecodecampster** [comment]
I just scanned it with Microsoft Defender on Windows 10 and it found no threats. Ran the program and it worked well 👌

**Comment by LittleTale** [comment]
https://www.virustotal.com/gui/file/F857CC534FE65C12B13E3D4892397EFCD04877364AC7BC68DC674D85B70728DB

now it detect alert from 2 sites... weird, is it because of the DLL ?

**Comment by schellingb** [comment]
These days it's pretty much impossible to avoid detection by certain scanners for a tiny tool like WiiBalanceScale given away for free.

In the old days anti virus software was built on signatures which detected actual malicious code. These days things are much different. A program that hasn't been signed with a 100$ (or more) code certificate is basically 90% guaranteed to get detected by certain scanners. If you look at that virustotal link and you look at the detections, they are named "W32.AIDetectMalware.CS", "Generic.ml", "Malicious.high.ml.score". The "ml" (machine learning) and "ai" (artificial (un)intelligence) there means it's just a fuzzy guess by some large (or small) model that most likely treats an unsigned application as very likely to be not "genuine" or not "official". Oh and then there's SecureAge, I don't know who makes that tool but I don't think I've ever seen it NOT detect something as malicious. Just trash...

So while we could spend some effort to fiddle with how our compiled code appears to these tools it's pretty much a lost fight. As long as it's just a small number of those machine learning detections I think most people can make a reasonable decision that it likely is just a false detection. If it's a major player like Microsoft Defender which is built into operating systems things are a bit different. As that will actually prevent many people from running the program at all, so it would be worth spending more effort there.

The code is free and open, people can read it and compile it themselves if they don't trust the binaries I provide. I can't give a guarantee that something I make in my spare time and give away for free won't cause your cat to get sick. So even if virustotal gave a perfect green result, without reading through the code and compiling it yourself, you still need to trust me :-)

**Comment by LittleTale** [comment]
> These days it's pretty much impossible to avoid detection by certain scanners for a tiny tool like WiiBalanceScale given away for free.
> 
> In the old days anti virus software was built on signatures which detected actual malicious code. These days things are much different. A program that hasn't been signed with a 100$ (or more) code certificate is basically 90% guaranteed to get detected by certain scanners. If you look at that virustotal link and you look at the detections, they are named "W32.AIDetectMalware.CS", "Generic.ml", "Malicious.high.ml.score". The "ml" (machine learning) and "ai" (artificial (un)intelligence) there means it's just a fuzzy guess by some large (or small) model that most likely treats an unsigned application as very likely to be not "genuine" or not "official". Oh and then there's SecureAge, I don't know who makes that tool but I don't think I've ever seen it NOT detect something as malicious. Just trash...
> 
> So while we could spend some effort to fiddle with how our compiled code appears to these tools it's pretty much a lost fight. As long as it's just a small number of those machine learning detections I think most people can make a reasonable decision that it likely is just a false detection. If it's a major player like Microsoft Defender which is built into operating systems things are a bit different. As that will actually prevent many people from running the program at all, so it would be worth spending more effort there.
> 
> The code is free and open, people can read it and compile it themselves if they don't trust the binaries I provide. I can't give a guarantee that something I make in my spare time and give away for free won't cause your cat to get sick. So even if virustotal gave a perfect green result, without reading through the code and compiling it yourself, you still need to trust me :-)

It's sad that it has come to this...
Pay or get flag basically... it sounds more and more like a subscription !
